### PR TITLE
vitess: init at 15.0.0

### LINKED ISSUE #146408: Packaging request: Vitess
**Project description**

Vitess is a database clustering system for horizontal scaling of MySQL originally built at YouTube.

**Metadata**

* homepage URL: https://vitess.io/
* source URL: https://github.com/vitessio/vitess
* license: Apache License 2.0
* platforms: unix, linux, darwin

### PR DESCRIPTION
**Pull request by urandom2**
###### Description of changes

Fixes #146408

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by deejayem — COMMENTED** [review]
`vitess.passthru.tests` are failing on all platforms.

File: pkgs/development/tools/database/vitess/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  vendorHash = "sha256-1iOT2AIDfCFRi8gZTA3A9cAzi/S07BHp7urD+rN1Ylw=";
+
+  buildInputs = [ sqlite ];
+
+  subPackages = [ "go/cmd/..." ];
+
+  preCheck = "ln -s $GOPATH/bin .";
+
+  meta = with lib; {
+    homepage = "http://vitess.io";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/database/vitess/default.nix:23]:
```suggestion
    homepage = "https://vitess.io/";
```

**Comment by SuperSandro2000** [comment]
> `vitess.passthru.tests` are failing on all platforms.

```
    vttestserver_test.go:64: 
                Error Trace:    vttestserver_test.go:64
                Error:          Received unexpected error:
                                fork/exec bin/mysqlctl: no such file or directory
                Test:           TestRunsVschemaMigrations
```

**Comment by urandom2** [comment]
> `vitess.passthru.tests` are failing on all platforms.

so, we do have to explicitly symlink link `$GOPATH/bin` to the build root, and test do pass during normal compilation. is there anything special I need to do to make `passthru.tests` work? how do I run them locally?

**Comment by deejayem** [comment]
Result of `nixpkgs-review pr 203905` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vitess</li>
  </ul>
</details>

**Review by deejayem — APPROVED** [review]
Built on aarch64-linux and x86_64-darwin, and tested that all of the binaries are able to print a help message (didn't test beyond that).

File: pkgs/development/tools/database/vitess/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  buildInputs = [ sqlite ];
+
+  subPackages = [ "go/cmd/..." ];
+
+  # integration tests require access to syslog and root
+  doCheck = false;
+
+  meta = with lib; {
+    homepage = "http://vitess.io/";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/database/vitess/default.nix:24]:
```suggestion
    homepage = "https://vitess.io/";
```

**Pull request merged**